src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -597,11 +597,13 @@ public boolean hasTag(String tag) {
 					}
 				}
 				if (!canParse) {
+					Skript.error(expr + " " + Language.get("is") + " " + notOfType(types), ErrorQuality.NOT_AN_EXPRESSION);
 					log.printError();
 					return null;
 				}
 				Parser<?> classInfoParser = classInfo.getParser();
 				if (classInfoParser == null || !classInfoParser.canParse(context)) {
+					Skript.error("A '" + expr  + "' cannot be parsed.");
```

Comment:
some tests to ensure these errors are working properly would be nice

Reply:
Also wait, it shouldn't be `expr` in that second error, it should be the classinfo.